orchagent/portsorch.cpp

```diff
@@ -1194,7 +1194,8 @@ bool PortsOrch::getPortMtu(const Port& port, sai_uint32_t &mtu)
 
     mtu = attr.value.u32 - (uint32_t)(sizeof(struct ether_header) + FCS_LEN + VLAN_TAG_LEN);
 
-    if (isMACsecPort(port.m_port_id))
+    /* Reduce the default MTU got from ASIC by MAX_MACSEC_SECTAG_SIZE */
```

Comment:
I don't think its correct to reduce the MTU generically which will affect non macsec scenarios. Ideally the fix here should be to avoid setting MTU when it is greater than max value.

Reply:
@dgsudharsan, the basic idea of this fix is to reduce the MTU given by SAI layer with the MACSEC sectag size in the API getPortMtu() which is called only in initializePort() which is called in switch boot up. If you check line :1195 in above code, we have a similar code to reduce the VLAN_TAG_LEN from MTU, even though all frames need not be tagged packets.

This fix is very localized in an API which is called during InitPort. This MTU value will be **overwritten/configured** later at https://github.com/sonic-net/sonic-swss/blob/master/orchagent/portsorch.cpp#L3679, based on the DB events from configured MTU value. 

Currently there is no way/SAI attribute to check the max MTU supported by SAI of an ASIC family. Could you share what mellanox ASIC/platforms would return as default MTU value when a get API of SAI_PORT_ATTR_MTU is called during initializePort().

Reply:
Our SAI returns 1522